scripts/set_config.py

```diff
@@ -151,6 +155,9 @@ def create_config_obj(argv, log=None):
 
         config_obj.update_values(user_config_obj)
 
+    if config_dict is not None:
+        config_obj.update_values(config_dict)
```

Comment:
Tying back to my comment about naming and this function possibly being responsible for too many things, mutating a `Config` object as a side effect might not belong in a function called `create_config_file()`. A refactor to a number of simpler functions would probably help here.

Reply:
Let's definitely have some open discussions about how to consider refactoring this tool and our others. Many of them follow a similar pattern.